lib/ops/opsservice/update.go

```diff
@@ -526,13 +526,66 @@ func (s *site) checkUpdateParameters(update *pack.PackageEnvelope, provisioner s
 			networkType, updateNetworkType)
 	}
 
-	if err = s.validateDockerConfig(*updateManifest); err != nil {
+	err = s.validateDockerConfig(*updateManifest)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	err = s.validateStorageConfig(*updateManifest)
+	if err != nil {
 		return trace.Wrap(err)
 	}
 
 	return nil
 }
 
+// validateStorageConfig makes sure that persistent storage configuration
+// in the new version is compatible with the old version.
+func (s *site) validateStorageConfig(updateManifest schema.Manifest) error {
+	installedEnabled := s.app.Manifest.OpenEBSEnabled()
+	updateEnabled := updateManifest.OpenEBSEnabled()
+
+	// OpenEBS wasn't enabled before, and isn't enabled in the new version,
+	// nothing to do.
+	if !installedEnabled && !updateEnabled {
+		return nil
+	}
+
+	// OpenEBS wasn't enabled before, but enabled in the new version, no
+	// specific checks are required.
+	if !installedEnabled && updateEnabled {
+		// TODO(r0mant): Check if OpenEBS was installed "out of band" and bail out?
+		return nil
+	}
+
+	// OpenEBS was enabled before, but disabled in the new version, we don't
+	// support uninstalling it at the moment.
+	if installedEnabled && !updateEnabled {
+		return trace.BadParameter(`The cluster has OpenEBS integration enabled but it's disabled in the version you're trying to upgrade to.
```

Comment:
This means rebuilding the cluster image. Can we make it a warning and continue as if it was not disabled?

Reply:
We sure can, but I'm not sure we can safely make the assumption that the updated application will work properly if it says that OpenEBS should be disabled but it's, in fact, enabled. We'll also end up with discrepancy b/w the declared state (manifest) and the actual state.